src/test/java/com/sda/microblogging/controller/PostControllerTest.java

```diff
@@ -0,0 +1,150 @@
+        post.setId(1);
+
+        ResultActions result = mockMvc
+                .perform(
+                        get("/home/public/post"))
+                .andDo(print());
+        result
+                .andExpect(status().isFound())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.*", hasSize(1)))
+                .andReturn();
+    }
+
+    @SneakyThrows
+    @Test
+    public void save_post_return_ok() {
+        PostSaveDTO postSaveDTO = new PostSaveDTO();
+        postSaveDTO.setContent("asa");
+        postSaveDTO.setCreationDate(Date.valueOf("2020-01-01"));
+        User user = new User();
+        user.setUserId(1);
+        postSaveDTO.setOwner(user);
+
+        ResultActions result = mockMvc
+                .perform(
+                        post("/home/post/create")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(asJsonString(user))
+                                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print());
+
+        result
+                .andExpect(status().isOk())
+                .andReturn();
+    }
+
+    @Test
+    public void get_posts_with_username() throws Exception {
+
+
+        ResultActions resultActions = mockMvc.perform(get("/home/posts/NotExists"))
+                .andExpect(status().isNotFound());
+        resultActions
+                .andExpect(jsonPath("$.*",hasSize(0)))
+                .andExpect(status().isOk())
+                .andReturn();
+    }
+
+    @Test
+    public void post_like() throws Exception{
+        PostLike postLike=  new PostLike();
+        Post p = new Post();
+        p.setId(1);
+        User u = new User();
+        u.setUserId(1);
+        postLike.setUser(u);
+        postLike.setPost(p);
+
+
+        RequestBuilder requestBuilder;
+        ResultActions resultActions = mockMvc.perform(post("/home/post/like")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(asJsonString(postLike)))
+                .andDo(print());
+
+        resultActions
+                .andExpect(status().isOk());
+    }
+    @Test
+    public void post_share_return_bad_request() throws Exception{
+
+
+        RequestBuilder requestBuilder;
+        ResultActions resultActions = mockMvc.perform(post("/home/post/like")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(asJsonString(null)))
+                .andDo(print());
+
+        resultActions
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void find_posts_by_users() throws Exception{
+
+
+        List<Post> postDTOS = Arrays.asList(postMapper.convertDtoToPost(new PostSaveDTO()));
+
+        when(postService.findAllPostsAndMyFollowingsPost(1)).thenReturn(postDTOS);
+
+        ResultActions resultActions = mockMvc.perform(get("/home/userId")
+                .param("userId","0")
```

Comment:
this param is about absolutely another thing: web request parameter, nothing about test data
like: /books?category=java 
so .param("category", "java")
And test logs explicitly show this:
MockHttpServletRequest:
      HTTP Method = GET
      Request URI = /home/userId 
so it does not get replaced..